src/instana/instrumentation/aioamqp.py

```diff
@@ -26,11 +26,31 @@ async def basic_publish_with_instana(
             "aioamqp-publisher", span_context=parent_context
         ) as span:
             try:
-                span.set_attribute("aioamqp.exchange", argv[0])
-                return await wrapped(*argv, **kwargs)
+                span.set_attribute("amqp.message", argv[0])
+                span.set_attribute("amqp.command", "publish")
+                span.set_attribute("amqp.exchange_name", kwargs.get("exchange_name"))
+                span.set_attribute("amqp.routing_key", kwargs.get("routing_key"))
+
+                protocol = getattr(instance, "protocol", None)
+                transport = getattr(protocol, "_transport", None)
+                extra = getattr(transport, "_extra", {}) if transport else {}
+                peername = extra.get("peername")
+                if (
+                    peername
+                    and isinstance(peername, (list, tuple))
+                    and len(peername) >= 2
+                ):
+                    connection_info = f"{peername[0]}:{peername[1]}"
+                else:
+                    connection_info = "unknown"
+                span.set_attribute("amqp.connection", connection_info)
+
+                response = await wrapped(*argv, **kwargs)
             except Exception as exc:
                 span.record_exception(exc)
                 logger.debug(f"aioamqp basic_publish_with_instana error: {exc}")
+            finally:
```

Comment:
You must use `else` here. 

The `finally` block executes if both the `try` and  `exception` blocks are executed. However, you don't set the `response` variable in the `exception` block, so it will raise an `UnboundLocalError`.

The `else` block executes only if the `try` block executes with no errors.